gateway/api_loader.go

```diff
@@ -676,10 +676,12 @@ func loadApps(specs []*APISpec) {
 
 	muxer := &proxyMux{}
 
-	globalConf := config.Global()
 	r := mux.NewRouter()
-	muxer.setRouter(globalConf.ListenPort, "", r)
+	r.NotFoundHandler = http.HandlerFunc(muxer.handle404)
+
+	globalConf := config.Global()
 	if globalConf.ControlAPIPort == 0 {
+		muxer.setRouter(globalConf.ListenPort, "", r)
```

Comment:
Why did you move this?  I think you are breaking the logic here.

So, `r` is bound on `globalConf.ListenPort` by default, if we don't have a separate `globalConf.ControlAPIPort` we will load control endpoints on `r` . When that control port is provided then we bind control endpoints there.

Please leave `muxer.setRouter(globalConf.ListenPort, "", r)` above the `if` condition as it was before.

Reply:
If it enter if, there is nothing change.
if it enters else statement, It is overriden in else statement. 

So I think it doesn't break.

Reply:
Let me make it more simple.